### PR TITLE
Allow configuration of URI style for cubes

### PR DESCRIPTION
**Pull request by goofballLogic**
Part of #378

### CODE REVIEW

File: csvcubed/csvcubed/writers/skoscodelistwriter.py
Comment (robons) [inline csvcubed/csvcubed/writers/skoscodelistwriter.py:155]: This needs to point at the true path of the CSV file, not the URI that we are using as an identifier for it.

Also on a side-note: `_get_identifier_for_document` is (by convention) a private function since it starts with a `_`.

File: csvcubed/tests/behaviour/steps/qbwriter.py
Comment (robons) [inline csvcubed/tests/behaviour/steps/qbwriter.py:808]: You can do set comprehensions just like list comprehensions in python, i.e. switch the `[ ... ]` for `{ ... }` and you don't need to convert to a set.

File: csvcubed/tests/behaviour/steps/qbwriter.py
Comment (robons) [inline csvcubed/tests/behaviour/steps/qbwriter.py:774]: I see what you've done here but I do worry these whitelists will make the tests fairly brittle if and when we add new RDF into the CSV-Ws in future. 

I think the way I would have approached this problem is to look for *relative URIs* which are defined as part of the graph - these are the new resources we're creating which use the newly generated URIs.

RDFlib isn't happy with relative URIs so it converts them to absolute URIs using the `metadataFilePath` in this instance. Now we've had some problems with the way that rdflib converts file paths into URIs (where Jenkins puts things like `@2` in folder names/paths), so I'd recommend setting `publicID="http://prefix-for-relative-uris"` in your call to `g.parse` which would mean that all of your relative URIs would end up with `http://prefix-for-relative-uris` prepended. Then you can filter by URIs that begin with that and ensure that they all either have or don't have the `.csv` extension on the end.

-------

Btw, `metadataFilePath` betrays your allegiance for other languages ;) would be `metadata_file_path` by convention in python.

File: csvcubed/tests/behaviour/cube.feature
Comment (robons) [inline csvcubed/tests/behaviour/cube.feature:31]: Yeah, this test isn't my favourite approach, sorry about the brittle nature.

**Review by robons — COMMENTED** [review]
I'm impressed you managed to get this far without more support.

I've left some comments. Let me know if you need help understanding anything.

File: csvcubed/tests/behaviour/steps/qbwriter.py
Comment (goofballLogic) [reply csvcubed/tests/behaviour/steps/qbwriter.py:808]: ooo nice

File: csvcubed/tests/behaviour/steps/qbwriter.py
Comment (goofballLogic) [reply csvcubed/tests/behaviour/steps/qbwriter.py:774]: ah damn i missed one - was a few days in when i noticed the snake casing

Comment (goofballLogic) [reply csvcubed/tests/behaviour/steps/qbwriter.py:774]: ok will switch to that approach. I noticed it was the relative ones I wanted but it felt a bit dirty looking for what rdflib happend to convert them too. Using publicID makes it a bit better.

File: csvcubed/csvcubed/models/cube/qb/components/dimension.py
Comment (goofballLogic) [reply csvcubed/csvcubed/models/cube/qb/components/dimension.py]: done

File: csvcubed/csvcubed/writers/urihelpers/skoscodelist.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/urihelpers/skoscodelist.py]: fixed

File: csvcubed/csvcubed/writers/qbwriter.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/qbwriter.py]: done

File: csvcubed/csvcubed/writers/qbwriter.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/qbwriter.py]: fixed

File: csvcubed/csvcubed/writers/qbwriter.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/qbwriter.py]: fixed

File: csvcubed/csvcubed/writers/qbwriter.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/qbwriter.py]: fixed

File: csvcubed/csvcubed/writers/urihelpers/skoscodelist.py
Comment (robons) [inline csvcubed/csvcubed/writers/urihelpers/skoscodelist.py:35]: I think I'd prefer `if self.code_list is not None and self.code_list.uri_style is not None` or the simplified `if self.code_list and self.code_list.uri_style` - I've been told (and believe) that negatives are harder to understand.

File: csvcubed/tests/behaviour/steps/qbwriter.py
Comment (robons) [inline csvcubed/tests/behaviour/steps/qbwriter.py:1]: I hope we don't need this import.

File: csvcubed/tests/unit/writers/test_qbwriter.py
Comment (robons) [inline csvcubed/tests/unit/writers/test_qbwriter.py:525]: This may well cause some problems on Jenkins if the file/folder path contains chars like `@`, etc. could you use a `publicID` when you parse instead so we can be sure to side-step the problem?

**Review by robons — APPROVED** [review]
I'll call it approved if you fix the couple of issues I'd raised. We shouldn't need another review I don't think.

File: csvcubed/csvcubed/writers/urihelpers/skoscodelist.py
Comment (goofballLogic) [reply csvcubed/csvcubed/writers/urihelpers/skoscodelist.py:35]: fixed

**Pull request merged**
